### PR TITLE
Fixes VSTS Bug 770474: NRE in ResultsEditorExtension

### PR DESCRIPTION
**Pull request by mkrueger**
https://devdiv.visualstudio.com/DevDiv/_workitems/edit/770474

### CODE REVIEW

**Comment by mkrueger** [comment]
@monojenkins backport release-7.8

File: main/src/addins/MonoDevelop.Refactoring/MonoDevelop.AnalysisCore/Gui/ResultsEditorExtension.cs [hunk 1]
```diff
@@ -342,91 +342,100 @@ static TextSegmentMarkerEffect GetSegmentMarkerEffect (IssueMarker marker)
...
+							newMarkers.Add (marker);
 						}
-						editor.AddMarker (marker);
-						newMarkers.Add (marker);
+						builder.Add (currentResult.QuickTask = new QuickTask (currentResult.Message, currentResult.Region.Start, currentResult.Level));
 					}
-					builder.Add (currentResult.QuickTask = new QuickTask (currentResult.Message, currentResult.Region.Start, currentResult.Level));
+					return true;
+				} catch (Exception ex) {
+					LoggingService.LogError ("Error while ResutsUpdater.IdleHandler", ex);
```
Comment (slluis) [inline main/src/addins/MonoDevelop.Refactoring/MonoDevelop.AnalysisCore/Gui/ResultsEditorExtension.cs]:
```suggestion
					LoggingService.LogInternalError ("Error while ResutsUpdater.IdleHandler", ex);
```

Comment (slluis) [reply main/src/addins/MonoDevelop.Refactoring/MonoDevelop.AnalysisCore/Gui/ResultsEditorExtension.cs]: It is an unexpected exception, so it should be reported as an internal error.

**Comment by slluis** [comment]
@monojenkins backport release-7.8

**Pull request merged**